cli/src/pcluster/validators/budget_validators.py

```diff
@@ -0,0 +1,46 @@
+from pcluster.aws.aws_api import AWSApi
+from pcluster.validators.common import FailureLevel, Validator
+
+
+class BudgetFilterTagValidator(Validator):
+    """
+    Budget CostFilters Tag validator.
+
+    Validate if the tag used for the filter is active under the user's account.
+    """
+
+    def _validate(self, budget_category):
+        budget_category_to_tag_key_map = {
+            "cluster": "parallelcluster:cluster-name",
+            "queue": "parallelcluster:queue-name",
+        }
+        self.tag_key = budget_category_to_tag_key_map.get(budget_category)
+        if self.tag_key is None:
+            return
+
+        self.cost_explorer_client = AWSApi.instance().cost_explorer
+
+        # Check if the tag is active
+        tag_status = self._get_tag_status()
+
+        if tag_status == "Inactive":
+            self._add_failure(
+                (
+                    f"Error: creating a budget of BudgetCategory: {budget_category} was not possible. "
+                    f"The tag: {self.tag_key} is inactive. Activate it and try again."
+                ),
+                FailureLevel.ERROR,
+            )
+
+        elif tag_status == "NotFound":
+            self._add_failure(
+                (
+                    f"Error: creating a budget of BudgetCategory: {budget_category} was not possible. "
+                    f"The tag: {self.tag_key} was not found in your account."
+                ),
+                FailureLevel.ERROR,
+            )
+
+    def _get_tag_status(self):
+        client_tags = self.cost_explorer_client.list_cost_allocation_tags()["CostAllocationTags"]
```

Comment:
nit: we don't need to pre-define `self.cost_explorer_client = AWSApi.instance().cost_explorer`, it is only used here.
We can simply use 
`client_tags = AWSApi.instance().cost_explorer.list_cost_allocation_tags()["CostAllocationTags"]` here

Reply:
Good insight. Updated!